core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -554,13 +554,13 @@ detray::rk_stepper<magnetic_field_t, transform3_t, constraint_t, policy_t,
                    inspector_t, array_t>::state::dqopds(const scalar_type qop)
     const -> typename transform3_t::scalar_type {
 
-    const auto& mat = *(this->_mat);
-
     // d(qop)ds is zero for empty space
     if (this->_mat == nullptr) {
         return 0.f;
     }
 
+    const auto& mat = *(this->_mat);
```

Comment:
alternatively there could be a helper function returning a reference to the material

```cpp
const auto& mat() const { return *_mat; }
```

Reply:
Good point, will add